ios-apps-master/Beer_Diary/Beer_Diary/Beer_List.swift

```diff
@@ -0,0 +1,221 @@
+//
+//  Beer_List.swift
+//  Beer_Diary
+//
+//  Created by Piotr Mielcarzewicz on 13/11/16.
+//  Copyright © 2016 Piotr Mielcarzewicz. All rights reserved.
+//
+
+import UIKit
+
+var BeerList = [BeerItem]()
+var detailed = BeerItem()
+
+class Beer_List: UITableViewController {
+    
+    let searchController = UISearchController(searchResultsController: nil)
+    var filteredBeers = [BeerItem]()
+    
+    func filterContentForSearchText(searchText: String, scope: String = "All")
+    {
+        filteredBeers = BeerList.filter{ beer in
+            return beer.name.lowercased().contains(searchText.lowercased())
+        }
+        tableView.reloadData()
+    }
+    
+
+    override func viewDidLoad() {
+        super.viewDidLoad()
+
+        if UserDefaults.standard.object(forKey: "name") != nil
+        {
+            var tname = [String]()
+            var tbrewery = [String]()
+            var tstyle = [String]()
+            var tcountry = [String]()
+            var tthoughts = [String]()
+            var taroma = [Int]()
+            var tappearance = [Int]()
+            var tpalate = [Int]()
+            var ttaste = [Int]()
+            var tscore = [Float]()
+            var tdate = [NSDate]()
+            
+            tname = UserDefaults.standard.object(forKey: "name") as! [String]
+            tbrewery = UserDefaults.standard.object(forKey: "brewery") as! [String]
+            tstyle = UserDefaults.standard.object(forKey: "style") as! [String]
+            tcountry = UserDefaults.standard.object(forKey: "country") as! [String]
```

Comment:
should not be `as!`. Should be `as?` and should be checked and error should be reported if cast failed anywhere